pages/index.tsx

```diff
@@ -0,0 +1,169 @@
+import Head from "next/head";
+import styles from "../styles/Home.module.css";
+import React, { useEffect, useRef, useState } from "react";
+import fetch from "node-fetch";
+import * as jwt from "jsonwebtoken";
+import {
+  CLIENT_ID,
+  HOSTED_UI_LOGIN_URL,
+  HOSTED_UI_TOKEN_URL,
+  REDIRECT_URI,
+} from "./api/constants";
+import extractUrlParameter from "./api/extractUrlParameter";
+import generateSaml from "./api/generateSaml";
+import {
+  DecodedIdToken,
+  IdDetails,
+  SsoDetails,
+  TokenResponse,
+} from "./api/types";
+import CallCenterPicker from "../components/CallCenterPicker";
+import SelfSubmittingSsoForm from "../components/SelfSubmittingSsoForm";
+
+export default function Home() {
+  const [idDetails, setIdDetails] = useState<IdDetails | undefined>(undefined);
+  const [ssoDetails, setSsoDetails] = useState<SsoDetails | undefined>(
+    undefined,
+  );
+  const [pickerSubmitWasPressed, setPickerSubmitWasPressed] = useState(false);
+  const [singleSubmitTriggered, setSingleSubmitTriggered] = useState(false);
```

Comment:
Feels like we could simplify these 2 into a single state variable, like with an enum "IDLE | USER_SUBMITTED | AUTO_TRIGGERED"

Reply:
Good point! I think given the current implementation, I could just use one boolean for both, but the enum allows us to possibly collect analytics in the future.